apps/indexer/test/indexer/fetcher/internal_transaction_test.exs

```diff
@@ -308,5 +308,111 @@ defmodule Indexer.Fetcher.InternalTransactionTest do
 
       assert %{block_hash: block_hash} = Repo.get(PendingBlockOperation, block_hash)
     end
+
+    test "handles problematic blocks correctly" do
```

Comment:
This test doesn't cover the case when a block has multiple transactions and internal transactions fetching failed for one of the transactions in the block only. Thus, this block should leave in pending_block_operations table for re-fetching.